### PR TITLE
Player Movement and Physics, 3rd Person Camera, and Debug Console

### PR DESCRIPTION
**Pull request by Rorazoro**

### CODE REVIEW

File: Assets/Scripts/ThirdPersonController.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+[RequireComponent (typeof (CharacterController))]
+public class ThirdPersonController : MonoBehaviour {
+    private CharacterController _controller;
+    private Vector2 _moveInput = new Vector2 ();
+    private Vector2 _lookInput = new Vector2 ();
+    private Quaternion nextRotation;
+    private Vector3 nextPosition;
```
Comment (moonbuffalo) [inline Assets/Scripts/ThirdPersonController.cs]: I dont see these being used, so I'd say get rid of them but not a huge deal

File: Assets/Scripts/ThirdPersonController.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        //reset the y rotation of the look transform
+        followTransform.transform.localEulerAngles = new Vector3 (angles.x, 0, 0);
+
+        //Calculate vertical speed and gravity
+        if (_controller.isGrounded) {
+            verticalSpeed = 0;
+        }
+
+        verticalSpeed -= Gravity;
+        position.y = verticalSpeed;
```
Comment (moonbuffalo) [inline Assets/Scripts/ThirdPersonController.cs]: Again not a huge deal but I don't really understand this. Line 86 looks right, if it's a speed, modifying it by gravity makes sense. but line 87 implies it's a position not a speed.

File: Assets/Scripts/ThirdPersonController.cs (hunk 1 above)
Comment (Rorazoro) [reply Assets/Scripts/ThirdPersonController.cs]: Old code. Removing.

File: Assets/Scripts/ThirdPersonController.cs (hunk 2 above)
Comment (Rorazoro) [reply Assets/Scripts/ThirdPersonController.cs]: I guess verticalMovement would make more sense. I expect more factors in the future to contribute to y movement. Jumping for example.

**Pull request merged**